### PR TITLE
[tests-only][full-ci] api test to set quota by users with different roles

### PR DESCRIPTION
**Pull request by PrajwolAmatya**
## Description
This PR adds API test to set quota by users with different roles

## Related Issue
Related to issue: https://github.com/owncloud/ocis/issues/5565

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiSpaces/setQuota.feature [hunk 1]
```diff
@@ -0,0 +1,183 @@
+@api @skipOnOcV10
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpaces/setQuota.feature:1]: We will remove it shortly so IMO, no need to add it. CC @grgprarup 
```suggestion
@api
```

File: tests/acceptance/features/apiSpaces/setQuota.feature [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+  As a user
+  I want to set quota to different users
+
+  Background:
+    Given these users have been created with default attributes and without skeleton files:
+      | username |
+      | Alice    |
+      | Brian    |
+
+  Scenario Outline: admin or space admin sets personal space quota of user with different role
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpaces/setQuota.feature:12]: one extra new line for tags
```suggestion


  Scenario Outline: admin or space admin sets personal space quota of user with different role
```

File: tests/acceptance/features/apiSpaces/setQuota.feature [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+      | Guest | Admin       |
+
+
+  Scenario Outline: admin or space admin user with different space role set quota of a space
+    Given the administrator has given "Alice" the role "Space Admin" using the settings api
+    And the administrator has given "Brian" the role "<userRole>" using the settings api
+    And user "Alice" has created a space "Project Jupiter" of type "project" with quota "20"
+    And user "Alice" has shared a space "Project Jupiter" with settings:
+      | shareWith | Brian  |
+      | role      | <spaceRole> |
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpaces/setQuota.feature:98]:
```suggestion
      | shareWith | Brian       |
      | role      | <spaceRole> |
```

**Review by SagarGi — COMMENTED** [review]
other LGTM :+1:

File: tests/acceptance/features/apiSpaces/setQuota.feature [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+      | User  | User        |
+      | User  | Guest       |
+      | User  | Admin       |
+      | Guest | Space Admin |
+      | Guest | User        |
+      | Guest | Guest       |
+      | Guest | Admin       |
+
+
+  Scenario Outline: admin or space admin user with different space role set quota of a space
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpaces/setQuota.feature:93]: `Admin` or `Space Admin` can edit any space regardless of their role in that space, isn't it?
IMO, `<spaceRole>` doesn't matter here. But you could add another scenario where a user with `manger`, `viewer`, `editor` roles edits the quota.

CC @amrita-shrestha @SwikritiT @SagarGi

Comment (PrajwolAmatya) [reply tests/acceptance/features/apiSpaces/setQuota.feature:93]: Scenario for `non-admin user` with `manager`, `viewer` and `editor` role trying to edit quota is on L140.

Comment (saw-jan) [reply tests/acceptance/features/apiSpaces/setQuota.feature:93]: Ahh, yeah. That should cover the space-role part.
I don't know what to do here since admin or space-admin user should be able to edit the quota of any space (without being a member of that space). What I think is assigning `<spaceRole>` is not necessary here but this might also cover other edge cases as well

Comment (PrajwolAmatya) [reply tests/acceptance/features/apiSpaces/setQuota.feature:93]: We could add another scenario where `admin` or `space-admin` user can edit the quota of any spaces.

File: tests/acceptance/features/apiSpaces/setQuota.feature [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+      }
+    }
+    """
+    Examples:
+      | userRole    |
+      | Admin       |
+      | Space Admin |
+
+
+  Scenario Outline: non-admin user with different space role tries to set quota of a space
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiSpaces/setQuota.feature:133]:
```suggestion
  Scenario Outline: User with role User and Guest can't set the quota of a space
```

File: tests/acceptance/features/apiSpaces/setQuota.feature [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+      | Admin       | Space Admin |
+      | Admin       | User        |
+      | Admin       | Guest       |
+      | Space Admin | Admin       |
+      | Space Admin | Space Admin |
+      | Space Admin | User        |
+      | Space Admin | Guest       |
+
+
+  Scenario Outline: non-admin user tries to set personal space quota of user with different role
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiSpaces/setQuota.feature:53]:
```suggestion
  Scenario Outline: User with role User and Guest can't set personal space quota of user with a different role
```

**Comment by amrita-shrestha** [comment]
@PrajwolAmatya please check if there is a scenario that covers setting quota multiple times. And any API to get quota information.

**Comment by PrajwolAmatya** [comment]
> @PrajwolAmatya please check if there is a scenario that covers setting quota multiple times. And any API to get quota information.

I couldn't find specific API to get quota information but quota information can be received while getting information of the spaces.

File: tests/acceptance/features/apiSpaces/setQuota.feature [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+    Examples:
+      | role        | userRole    |
+      | Admin       | Admin       |
+      | Admin       | Space Admin |
+      | Admin       | User        |
+      | Admin       | Guest       |
+      | Space Admin | Admin       |
+      | Space Admin | Space Admin |
+      | Space Admin | User        |
+      | Space Admin | Guest       |
```
[the remark this answers is not in the document]
Comment (PrajwolAmatya) [reply tests/acceptance/features/apiSpaces/setQuota.feature]: yes, i have updated the scenarios.

**Review by saw-jan — COMMENTED** [review]
LGTM

File: tests/acceptance/features/apiSpaces/setQuota.feature [hunk 8]
```diff
@@ -0,0 +1,177 @@
+@api
+Feature: Set quota
+  As a user
+  I want to set quota to different users
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiSpaces/setQuota.feature:4]: So that where

File: tests/acceptance/features/apiSpaces/setQuota.feature [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+    """
+    Examples:
+      | role  | userRole    |
+      | Admin | Admin       |
+      | Admin | Space Admin |
+      | Admin | User        |
+      | Admin | Guest       |
+
+
+  Scenario Outline: User with role Space Admin, User and Guest can't set personal space quota of user with a different role
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiSpaces/setQuota.feature:49]:
```suggestion
  Scenario Outline: user other than admin can't set the personal space quota of other users 
```

**Review by amrita-shrestha — COMMENTED** [review]
others lgtm

File: tests/acceptance/features/apiSpaces/setQuota.feature [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+    }
+    """
+    Examples:
+      | userRole | spaceRole |
+      | User     | viewer    |
+      | User     | editor    |
+      | User     | manager   |
+      | Guest    | viewer    |
+      | Guest    | editor    |
+      | Guest    | manager   |
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiSpaces/setQuota.feature:178]: does space role matter while testing this scenario?

File: tests/acceptance/features/apiSpaces/setQuota.feature [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+
+  Background:
+    Given these users have been created with default attributes and without skeleton files:
+      | username |
+      | Alice    |
+      | Brian    |
+
+
+  Scenario Outline: admin sets personal space quota of user with different role
+    Given the administrator has given "Alice" the role "<role>" using the settings api
```
Comment (grgprarup) [inline tests/acceptance/features/apiSpaces/setQuota.feature:15]:
```suggestion
    Given the administrator has given "Alice" the role "Admin" using the settings api
```

File: tests/acceptance/features/apiSpaces/setQuota.feature [hunk 12]
```diff
@@ -0,0 +1,178 @@
...
+        }
+      }
+    }
+    """
+    Examples:
+      | role  | userRole    |
+      | Admin | Admin       |
+      | Admin | Space Admin |
+      | Admin | User        |
+      | Admin | Guest       |
```
Comment (grgprarup) [inline tests/acceptance/features/apiSpaces/setQuota.feature:47]:
```suggestion
      | userRole    |
      | Admin       |
      | Space Admin |
      | User        |
      | Guest       |
```

File: tests/acceptance/features/apiSpaces/setQuota.feature [hunk 13]
```diff
@@ -0,0 +1,178 @@
...
+      | User        | Space Admin |
+      | User        | User        |
+      | User        | Guest       |
+      | Guest       | Admin       |
+      | Guest       | Space Admin |
+      | Guest       | User        |
+      | Guest       | Guest       |
+
+
+  Scenario Outline: admin or space admin user sets quota of a space
```
Comment (grgprarup) [inline tests/acceptance/features/apiSpaces/setQuota.feature:94]:
```suggestion
  Scenario Outline: admin or space admin user sets a quota of a project space
```
Seems easier to figure out whether a scenario is about personal or project space.

File: tests/acceptance/features/apiSpaces/setQuota.feature (hunk 10 above)
Comment (grgprarup) [inline tests/acceptance/features/apiSpaces/setQuota.feature:178]:
```suggestion
      | userRole |
      | User     |
      | Guest    |
```

Comment (saw-jan) [reply tests/acceptance/features/apiSpaces/setQuota.feature]: Could add a scenario where a user (who is a member of space with <role>) tries to edit quota

File: tests/acceptance/features/apiSpaces/setQuota.feature [hunk 14]
```diff
@@ -0,0 +1,220 @@
...
+            }
+          }
+        }
+      }
+    }
+    """
+    Examples:
+      | userRole |
+      | User     |
+      | Guest    |
```
Comment (grgprarup) [inline tests/acceptance/features/apiSpaces/setQuota.feature:220]:
```suggestion
```
This will cover in the line 134 scenario.

File: tests/acceptance/features/apiSpaces/setQuota.feature [hunk 15]
```diff
@@ -0,0 +1,178 @@
...
+      }
+    }
+    """
+    Examples:
+      | userRole    |
+      | Admin       |
+      | Space Admin |
+
+
+  Scenario Outline: users with role User and Guest having space role can't set the quota of a space
```
Comment (SwikritiT) [inline tests/acceptance/features/apiSpaces/setQuota.feature:134]:
```suggestion
  Scenario Outline: users with role User and Guest can't set the quota of a space
```
would this also make sense in this context? `having space role` is this really needed?

**Comment by ownclouders** [comment]
E2E tests failed: https://drone.owncloud.com/owncloud/ocis/20782/86/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/ocis/20782/tracing/admin-deletes-group-admin-2023-3-24-05-00-38.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/ocis/20782/tracing/unstructured-collection-of-testable-space-interactions-alice-2023-3-24-05-11-02.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/ocis/20782/tracing/user-group-assignments-can-be-handled-via-batch-actions-admin-2023-3-24-05-01-27.zip 


</p></details>

File: tests/acceptance/features/apiSpaces/setQuota.feature [hunk 16]
```diff
@@ -0,0 +1,178 @@
...
+    """
+    Examples:
+      | userRole    |
+      | Admin       |
+      | Space Admin |
+      | User        |
+      | Guest       |
+
+
+  Scenario Outline: user other than admin can't set the personal space quota of other users
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpaces/setQuota.feature:50]:
```suggestion
  Scenario Outline: non-admin user tries to set the personal space quota of other users
```

File: tests/acceptance/features/apiSpaces/setQuota.feature [hunk 17]
```diff
@@ -0,0 +1,178 @@
...
+      }
+    }
+    """
+    Examples:
+      | userRole    |
+      | Admin       |
+      | Space Admin |
+
+
+  Scenario Outline: users with role User and Guest can't set the quota of a space
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpaces/setQuota.feature:134]:
```suggestion
  Scenario Outline: normal or guest user tries to set the quota of a space
```

**Review by saw-jan — APPROVED** [review]
LGTM

**Review by SwikritiT — APPROVED** [review]

**Review by SwikritiT — CHANGES_REQUESTED** [review]
```
This branch is [1 commit ahead](https://github.com/owncloud/ocis/compare/master...test-for-set-quota-for-differnt-users), [79 commits behind](https://github.com/owncloud/ocis/compare/test-for-set-quota-for-differnt-users...master) master.
```
Can you rebase first?

**Review by SwikritiT — APPROVED** [review]
LGTM if CI passes
